justfile

```diff
@@ -142,25 +149,50 @@ lint: clippy fmt-check
 
 package-dir:
     mkdir -p packages
+    rm -rf packages/prep
 
-[linux]
-package: build get-binary package-dir
-    tar cv {{output-filename}} | gzip -9 > "packages/cargo-binstall-{{target}}.tgz"
+package-prepare: build package-dir
+    just get-binary packages/prep
+    -just get-debuginfo {{output-filename}} packages/prep
+
+    just get-output detect-wasi{{output-ext}} packages/prep
+    -just get-debuginfo detect-wasi{{output-ext}} packages/prep
+
+    cp crates/bin/LICENSE packages/prep
+    cp README.md packages/prep
 
 [macos]
-package: build get-binary package-dir
-    zip -9 "packages/cargo-binstall-{{target}}.zip" {{output-filename}}
+lipo-prepare: package-dir
+    just target=aarch64-apple-darwin build get-binary packages/prep/arm64
+    just target=x86_64-apple-darwin build get-binary packages/prep/x64
+    lipo -create -output packages/prep/{{output-filename}} arm64/{{output-filename}} x64/{{output-filename}}
+    -just get-debuginfo {{output-filename}} packages/prep
 
-[windows]
-package: build get-binary package-dir
-    7z a -mx9 "packages/cargo-binstall-{{target}}.zip" {{output-filename}}
+    just target=aarch64-apple-darwin build get-output detect-wasi{{output-ext}} packages/prep/arm64
+    just target=x86_64-apple-darwin build get-output detect-wasi{{output-ext}} packages/prep/x64
+    lipo -create -output packages/prep/{{output-filename}} arm64/{{output-filename}} x64/{{output-filename}}
+    -just get-debuginfo detect-wasi{{output-ext}} packages/prep
+
+    cp crates/bin/LICENSE packages/prep
+    cp README.md packages/prep
+
+
+[linux]
+package: package-prepare
+    cd packages/prep && tar cv {{output-filename}} | gzip -9 > "../cargo-binstall-{{target}}.tgz"
```

Comment:
Could use:
```suggestion
    tar -C packages/prep cv {{output-filename}} | gzip -9 > "../cargo-binstall-{{target}}.tgz"
```

Reply:
I prefer doing it this way just because it's symmetrical to the other platforms' `package`, which don't support `-C`